### PR TITLE
feat: add ExternalContactGetCorpTagList,ExternalContactAddCorpTag,ExternalContactEditCorpTag,ExternalContactDelCorpTag,ExternalContactMarkTag

### PR DESCRIPTION
**Pull request by w3xse7en**
Support Manage ExternalContact CorpTag

ExternalContactGetCorpTagList doc: https://work.weixin.qq.com/api/doc/90000/90135/92117#获取企业标签库
ExternalContactAddCorpTag doc: https://work.weixin.qq.com/api/doc/90000/90135/92117#添加企业客户标签
ExternalContactEditCorpTag doc: https://work.weixin.qq.com/api/doc/90000/90135/92117#编辑企业客户标签
ExternalContactDelCorpTag doc: https://work.weixin.qq.com/api/doc/90000/90135/92117#删除企业客户标签
ExternalContactMarkTag doc: https://work.weixin.qq.com/api/doc/90000/90135/92118

### CODE REVIEW

File: docs/external_contact.md [hunk 1]
```diff
@@ -151,3 +151,33 @@ Name|JSON|Type|Doc
...
 `RemarkPicMediaid`|`remark_pic_mediaid`|`string`| 备注图片的mediaid，remark_pic_mediaid可以通过素材管理接口获得。
+
+### `ExternalContactCorpTag` 企业客户标签
+
+Name|JSON|Type|Doc
+:---|:---|:---|:--
+`ID`|`id`|`string`| 标签id
+`Name`|`name`|`string`| 标签名称
+`CreateTime`|`create_time`|`int`| 标签创建时间
+`Order`|`order`|`int`| 标签排序的次序值，order值大的排序靠前。有效的值范围是[0, 2^32)
```
Comment (xen0n) [inline docs/external_contact.md:162]: 既然范围是0到2^32-1，那么类型就应该是uint32

File: docs/external_contact.md [hunk 2]
```diff
@@ -151,3 +151,33 @@ Name|JSON|Type|Doc
...
+`Deleted`|`deleted`|`bool`| 标签是否已经被删除，只在指定tag_id进行查询时返回
+
+### `ExternalContactCorpTagGroup` 企业客户标签
+
+Name|JSON|Type|Doc
+:---|:---|:---|:--
+`GroupID`|`group_id`|`string`| 标签组id
+`GroupName`|`group_name`|`string`| 标签组名称
+`CreateTime`|`create_time`|`int`| 标签组创建时间
+`Order`|`order`|`int`| 标签组排序的次序值，order值大的排序靠前。有效的值范围是[0, 2^32)
```
Comment (xen0n) [inline docs/external_contact.md:172]: 同

File: README.md [hunk 3]
```diff
@@ -114,6 +114,11 @@ in at least 2 of Qiniu's internal systems.
     - [x] 获取客户列表
     - [x] 获取客户详情
     - [x] 修改客户备注信息
+* [x] 客户标签管理
+   - [x] 管理企业标签
+   - [x] 编辑客户企业标签
+
+
```
Comment (xen0n) [inline README.md:121]: remove these 2 empty lines

File: external_contact.go [hunk 4]
```diff
@@ -29,3 +29,55 @@ func (c *WorkwxApp) RemarkExternalContact(req *ExternalContactRemark) error {
 	})
 	return err
 }
+
+// GetExternalContactCorpTagList 获取企业标签库
+func (c *WorkwxApp) GetExternalContactCorpTagList(tagID ...string) ([]ExternalContactCorpTagGroup, error) {
```
Comment (xen0n) [inline external_contact.go:34]: `ListExternalContactCorpTags` for idiomatic English naming?

File: external_contact.go [hunk 5]
```diff
@@ -29,3 +29,55 @@ func (c *WorkwxApp) RemarkExternalContact(req *ExternalContactRemark) error {
...
+func (c *WorkwxApp) DelExternalContactCorpTag(tagID, groupID []string) error {
+	_, err := c.execExternalContactDelCorpTag(reqExternalContactDelCorpTag{
+		TagID:   tagID,
+		GroupID: groupID,
+	})
+	return err
+}
+
+// MarkExternalContactTag 标记客户企业标签
+func (c *WorkwxApp) MarkExternalContactTag(userID, externalUserID string, addTag, removeTag []string) error {
```
Comment (xen0n) [inline external_contact.go:75]: `EditExternalContactTags` might be better, but Tencent already used the "edit" verb for the CRUD endpoints above, so unfortunately we have to live with this "mark" thing. /sigh

**Review by xen0n — COMMENTED** [review]
Squash all the commits into one and you're ready to go!

**Comment by xen0n** [comment]
Looks good!

bors r+

File: external_contact.go (hunk 4 above)
Comment (w3xse7en) [reply external_contact.go]: at fisrt i named `ListExternalContactCorpTags` but reference url `get_corp_tag_list` i choose `GetExternalContactCorpTagList`,  luckly we see eye to eye

**Pull request merged**